packageurl_test.go

```diff
@@ -319,8 +319,18 @@ func TestEncoding(t *testing.T) {
 		},
 		{
 			name:     "unencoded namespace segment is encoded",
-			input:    "pkg:type/%3F%40name/spac%20e/name@version?key=value#sub/path",
-			expected: "pkg:type/%3F%40name/spac%20e/name@version?key=value#sub/path",
+			input:    "pkg:type/name/spac e/name@version?key=value#sub/path",
```

Comment:
Hope i did not misunderstand the test.
I think the input must be encoded as it goes though the `FromString` (decoding phase), then back to string (Encoding phase).
If the test tries to test both funcions the `input` and `expected` must both be properly purl encoding.
I not sure if this intentional (If so disregard this) but test probably works on these `input` un-encoded values, because the first `FromString` will not decode anything.